src/main/java/io/confluent/idesidecar/restapi/proxy/clusters/strategy/ConfluentLocalKafkaClusterStrategy.java

```diff
@@ -1,45 +1,64 @@
 package io.confluent.idesidecar.restapi.proxy.clusters.strategy;
 
+import static io.confluent.idesidecar.restapi.util.RequestHeadersConstants.CONNECTION_ID_HEADER;
+
+import io.confluent.idesidecar.restapi.proxy.clusters.ClusterProxyContext;
+import io.vertx.core.MultiMap;
+import io.vertx.core.http.HttpHeaders;
 import jakarta.enterprise.context.ApplicationScoped;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 
 /**
- * Strategy for processing requests and responses for a local Kafka cluster.
+ * Strategy for processing requests and responses for a Confluent local Kafka cluster.
  */
 @ApplicationScoped
 public class ConfluentLocalKafkaClusterStrategy extends ClusterStrategy {
 
-  final String confluentLocalKafkaRestHostname;
+  @ConfigProperty(name = "ide-sidecar.api.host")
+  String sidecarHost;
 
-  public ConfluentLocalKafkaClusterStrategy(
-      @ConfigProperty(name = "ide-sidecar.connections.confluent-local.default.kafkarest-hostname")
-      String confluentLocalKafkaRestHostname
-  ) {
-    this.confluentLocalKafkaRestHostname = confluentLocalKafkaRestHostname;
+  /**
+   * We require the connection ID header to be passed to our implementation of the Kafka REST API
+   * at the /internal/kafka path. This is used to identify the connection that the request is
+   * associated with.
+   * @param context The context of the proxy request.
+   * @return The headers to be passed to our implementation of the Kafka REST API.
+   */
+  @Override
+  public MultiMap constructProxyHeaders(ClusterProxyContext context) {
+    return HttpHeaders
+        .headers()
+        .add(CONNECTION_ID_HEADER, context.getConnectionId());
   }
 
-
+  /**
+   * Route the request back to ourselves at the /internal/kafka path.
+   * Context: We used to send this proxy request to the Kafka REST server running in the
+   * confluent-local container, but now we route the request to our own implementation of the
+   * Kafka REST API, served at the /internal/kafka path. This was done to get early feedback
+   * on our in-house implementation of the Kafka REST API.
```

Comment:
Why just "early feedback"? Is this not the longer term direction?

Reply:
Yes, it is the longer term direction. By "early feedback", I meant that if we simply add the Kafka REST implementation to ide-sidecar without having it be the backbone of an existing feature set (confluent local), then we'd only discover any latent issues/bugs (not covered by integration tests) once we actually _do_ start using it end-to-end from the VS Code extension.